### PR TITLE
fix: resolve regex with quantifiers

### PR DESCRIPTION
**Pull request by gcusnieux**
gravitee-io/issues#8217

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.9.5-2359-fix-regex-quantifiers-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/el/gravitee-expression-language/1.9.5-2359-fix-regex-quantifiers-SNAPSHOT/gravitee-expression-language-1.9.5-2359-fix-regex-quantifiers-SNAPSHOT.zip)

### CODE REVIEW

File: src/test/java/io/gravitee/el/spel/context/SpelTemplateEngineTest.java [hunk 1]
```diff
@@ -692,4 +693,36 @@ public void shouldGetValueAsBoolean() {
...
+
+        Pattern pattern = java.util.regex.Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(pathInfo);
+        assertEquals(true, matcher.matches());
+
+        when(request.pathInfo()).thenReturn(pathInfo);
+        final TemplateEngine engine = TemplateEngine.templateEngine();
+        engine.getTemplateContext().setVariable("request", new EvaluableRequest(request));
+
+        engine.getValue("{#request.pathInfo.matches('" + regex + "')}", Boolean.class);
```
Comment (gaetanmaisse) [inline src/test/java/io/gravitee/el/spel/context/SpelTemplateEngineTest.java:710]: Should we `assertTrue` for this call just to be sure it's working fine in an EL expression?

File: src/test/java/io/gravitee/el/spel/context/SpelTemplateEngineTest.java [hunk 2]
```diff
@@ -692,4 +693,36 @@ public void shouldGetValueAsBoolean() {
...
```
Comment (gaetanmaisse) [inline src/test/java/io/gravitee/el/spel/context/SpelTemplateEngineTest.java:726]: same here?

**Review by Okhelifi — APPROVED** [review]

**Pull request merged**